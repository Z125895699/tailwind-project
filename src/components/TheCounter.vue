<script setup lang="ts">
const props = defineProps<{
  initial: number
}>()

const { count, inc, dec } = useCounter(props.initial)
</script>

<template>
  <div>
    <span class="count-value">{{ count }}</span>
    <button class="inc" @click="inc()">
      +
    </button>
    <button class="dec" @click="dec()">
      -
    </button>
  </div>
</template>
