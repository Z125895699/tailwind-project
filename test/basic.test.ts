import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'

import TheCounter from '../src/components/TheCounter.vue'

describe('tests', () => {
  it('should works', () => {
    const wrapper = mount(TheCounter, { props: { initial: 10 } })
    console.log(wrapper)
    // expect(1 + 1).toMatchSnapshot()
    expect(1 + 1).toEqual(2)
  })
})
