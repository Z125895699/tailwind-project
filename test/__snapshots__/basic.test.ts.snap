// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TheCounter.vue > should render 1`] = `"<div><span class=\\"count-value\\">10</span><button class=\\"inc\\"> + </button><button class=\\"dec\\"> - </button></div>"`;
